samples/grid-pro/demo/validation/demo.js

```diff
@@ -0,0 +1,170 @@
+Grid.grid('container', {
+    dataTable: {
+        columns: {
+            productName: [
+                'Phone',
+                'Bread',
+                'Shirt',
+                'Laptop',
+                'Milk',
+                'Jeans',
+                'Tablet'
+            ],
+            sku: [
+                'ELE-001',
+                'FOO-123',
+                'CLO-456',
+                'ELE-002',
+                'FOO-456',
+                'CLO-789',
+                'ELE-003'
+            ],
+            category: [
+                'Electronics',
+                'Food',
+                'Clothing',
+                'Electronics',
+                'Food',
+                'Clothing',
+                'Electronics'
+            ],
+            price: [699.99, 10.5, 29.99, 1199.99, 12, 49.99, 500],
+            discount: [20, 5, 40, 25, 10, 15, 30]
+        }
+    },
+    columnDefaults: {
+        cells: {
+            editMode: {
+                enabled: true
+            }
+        }
+    },
+    columns: [{
+        id: 'productName',
+        dataType: 'string',
+        cells: {
+            editMode: {
+                validationRules: [
+                    'notEmpty',
+                    {
+                        validate: function ({ value }) {
+                            return /^[a-zA-Z0-9\s]+$/.test(value);
+                        },
+                        notification: 'No special characters allowed.'
+                    }
+                ]
+            }
+        },
+        header: {
+            format: 'Product name'
+        }
+    }, {
+        id: 'sku',
+        dataType: 'string',
+        cells: {
+            editMode: {
+                validationRules: [
+                    {
+                        validate: 'unique',
+                        notification: 'SKU must be unique.'
+
+                    }, {
+                        validate: function ({ value }) {
+                            const columnData = this.column.data;
+
+                            return !columnData.includes(value) &&
+                                /^[A-Z]{3}-\d{3}$/.test(value);
+                        },
+                        notification: 'SKU must be in the format AAA-123.'
+                    }
+                ]
+            }
```

Comment:
You don't need to check uniqueness two times.
```suggestion
                validationRules: [{
                    validate: 'unique',
                    notification: 'SKU must be unique.'
                }, {
                    validate: ({ rawValue }) => /^[A-Z]{3}-\d{3}$/.test(rawValue),
                    notification: 'SKU must be in the format AAA-123.'
                }]
```